File: typescript/src/models/componentParameters.ts
// this file is @generated
import { type BillingPeriodEnum, BillingPeriodEnumSerializer } from "./billingPeriodEnum";

export interface ComponentParameters {
  billingPeriod?: BillingPeriodEnum | null;

  committedCapacity?: number | null;

  initialSlotCount?: number | null;
}

export const ComponentParametersSerializer = {
  _fromJsonObject(object: any): ComponentParameters {
    return {
      billingPeriod:
        object["billing_period"] != null
          ? BillingPeriodEnumSerializer._fromJsonObject(object["billing_period"])
          : undefined,
      committedCapacity: object["committed_capacity"],
      initialSlotCount: object["initial_slot_count"],
    };
  },

  _toJsonObject(self: ComponentParameters): any {
    return {
      billing_period:
        self.billingPeriod != null
          ? BillingPeriodEnumSerializer._toJsonObject(self.billingPeriod)
          : undefined,
      committed_capacity: self.committedCapacity,
      initial_slot_count: self.initialSlotCount,
    };
  },
};

File: python/meteroid/models/component_parameters.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .billing_period_enum import BillingPeriodEnum


@dataclasses.dataclass
class ComponentParameters(BaseModel):
    billing_period: t.Optional[BillingPeriodEnum] = None

    committed_capacity: t.Optional[int] = None

    initial_slot_count: t.Optional[int] = None
